### PR TITLE
Consume the new ISymUnmanagedReader3.GetSymAttributeByVersion API

### PR DESCRIPTION
**Pull request by amcasey**
Not having this API was causing problems in EnC scenarios because (e.g.) the hoisted local scopes in state machine methods always came from the unmodified method (version 1).

### CODE REVIEW

**Comment by amcasey** [comment]
FYI @KevinH-MS @cston @ManishJayaswal

File: src/Compilers/CSharp/Test/Emit/Emit/EditAndContinue/LocalSlotMappingTests.cs [hunk 1]
```diff
@@ -246,13 +246,13 @@ .maxstack  2
   </methods>
 </symbols>
 ");
-            var debugInfoProvider = v0.CreatePdbInfoProvider();
+            var symReader = v0.CreateSymReader();
```
Comment (KevinH-MS) [inline src/Compilers/CSharp/Test/Emit/Emit/EditAndContinue/LocalSlotMappingTests.cs]: make "using (var symReader = ...) { ..."?

File: src/Compilers/CSharp/Test/Emit/Emit/EditAndContinue/EditAndContinueClosureTests.cs [hunk 2]
```diff
@@ -100,7 +100,7 @@ void F()
             var f0 = compilation0.GetMember<MethodSymbol>("C.F");
             var f1 = compilation1.GetMember<MethodSymbol>("C.F");
 
-            var generation0 = EmitBaseline.CreateInitialBaseline(md0, v0.CreatePdbInfoProvider().GetEncMethodDebugInfo);
+            var generation0 = EmitBaseline.CreateInitialBaseline(md0, v0.CreateSymReader().GetEncMethodDebugInfo);
```
Comment (KevinH-MS) [inline src/Compilers/CSharp/Test/Emit/Emit/EditAndContinue/EditAndContinueClosureTests.cs]: Should we be Disposing these?

Comment (tmat) [inline src/Compilers/CSharp/Test/Emit/Emit/EditAndContinue/EditAndContinueClosureTests.cs]: No need to dispose in tests. GC will collect them.

**Comment by tmat** [comment]
:+1:

File: src/Compilers/CSharp/Test/Emit/Emit/EditAndContinue/LocalSlotMappingTests.cs (hunk 1 above)
Comment (tmat) [inline src/Compilers/CSharp/Test/Emit/Emit/EditAndContinue/LocalSlotMappingTests.cs]: Let's not add usings - it makes the test unnecessarily verbose.

Comment (KevinH-MS) [inline src/Compilers/CSharp/Test/Emit/Emit/EditAndContinue/LocalSlotMappingTests.cs]: Then can we remote the .Dispose (assuming it doesn't change the observable behavior)?

**Comment by KevinH-MS** [comment]
LGTM

**Pull request merged**